### ISSUE TITLE
Add ability to track messages being displayed to users

### ISSUE THREAD

**Issue by rchiodo** [microsoft/pylance-release#4482]
We're about to localize all of our error messages (@bschnurr is currently working on this for pylance and pyright). 

We'd like to add telemetry to track what the most common error messages are so we can provide better documentation.

Is there some way to track what errors are being reported? 

I think we might be able to this in a diagnostic sink:
https://github.com/microsoft/pyrx/main/packages/pyright/packages/pyright-internal/src/common/diagnosticSink.ts#L78

but the localization of the strings makes identifying the message hard.

**Comment by rchiodo** [comment]
Submitted a change using the localization hook that tracks the 'DiagnosticsSeen'.

This comes as part of our analysis completed telemetry and looks something like this:

"DiagnosticsSeen": "Diagnostic.importResolveFailure:2;Diagnostic.importSourceResolveFailure:1"

/cc @luabud

**Comment by rchiodo** [comment]
To verify:

- Turn on Developer Log Level : Trace
- Open the 'Extension Telemetry' pane in the output
- See the 'DiagnosticsSeen' property in the analysis_complete message:

```
2023-07-26 14:55:33.831 [trace] ms-python.python/language_server/analysis_complete {"properties":{"lsVersion":"2023.7.41","DiagnosticsSeen":"Diagnostic.unaccessedSymbol:1;Diagnostic.unaccessedImport:1","abexp.assignmentcontext":"vsliv368:30146709;vsreu685:30147344;bridge0708:30335490;bridge0723:30353136;vsaa593:30376534;py29gd2263:30784851;pynewext54:30618038;pythonfmttext:30716741;pythoncmvfstr:30726892;pythongtdpath:30726887;gsofa:30797620;pythonnosmt12:30797651;pythonnoceb:30776497;","common.os":"win32","common.nodeArch":"x64","common.platformversion":"10.0.22621","common.telemetryclientversion":"0.7.7","common.extname":"ms-python.python","common.extversion":"2023.12.0","common.vscodemachineid":"21d7289ca11f451b6cb7b2ffdd63459a88ef948dbd3f7cc2494d6b08994757ea","common.vscodesessionid":"62b4d856-f3c1-429e-a0d6-cf79896eefa216
```

**Comment by rchiodo** [comment]
This issue has been fixed in prerelease version 2023.7.41, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#2023741-26-July-2023-PreRelease)
